File: eKucniLjubimci.Services/Database/Narudzba.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Text.Json.Serialization;
using System.Threading.Tasks;

namespace eKucniLjubimci.Services.Database
{
    public class Narudzba
    {
        public int NarudzbaId { get; set; }
        public DateTime DatumNarudzbe { get; set; }
        public decimal TotalFinal { get; set; }
        public string StateMachine { get; set; }

        
        public virtual Kupac Kupac { get; set; }
        public int KupacId { get; set; }

        public virtual List<NarudzbaArtikal> NarudzbeArtikli { get; set; } = new List<NarudzbaArtikal>();

        public virtual List<Zivotinja> Zivotinje { get; set; }=new List<Zivotinja>();

    }
}

File: flutterUI/flutter_mobile/lib/models/narudzba.dart
import 'package:flutter_mobile/models/zivotinja.dart';
import 'package:json_annotation/json_annotation.dart';

import 'kupac.dart';
import 'narudzba_artikal.dart';

part 'narudzba.g.dart';

@JsonSerializable()
class Narudzba {
  int? narudzbaId;
  String? datumNarudzbe;
  double? totalFinal;
  String? stateMachine;
  Kupac? kupac;
  int? kupacId;
  List<NarudzbaArtikal>? narudzbeArtikli;
  List<Zivotinja>? zivotinje;

  Narudzba({
    this.narudzbaId,
    this.datumNarudzbe,
    this.totalFinal,
    this.stateMachine,
    this.kupac,
    this.kupacId,
    this.narudzbeArtikli,
    this.zivotinje,
  });

  factory Narudzba.fromJson(Map<String, dynamic> json) =>
      _$NarudzbaFromJson(json);

  Map<String, dynamic> toJson() => _$NarudzbaToJson(this);
}
